### PR TITLE
Classification of algorithms

### PR DESCRIPTION
**Pull request by miguelgfierro**
### Description

The first iteration to start the discussion. Please share your comments.

After we decide the names, we also would need to update the glossary

### Checklist:

- [ ] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [ ] I have added tests covering my contributions.
- [ ] I have updated the documentation accordingly.
- [ ] This PR is being made to `staging branch` and not to `main branch`.

### CODE REVIEW

**Review by anargyri — COMMENTED** [review]
Why not keep the old labels too, they are informative. You can separate them with commas e.g. "collaborative filtering, sequential"

**Review by yueguoguo — APPROVED** [review]

**Comment by miguelgfierro** [comment]
@anargyri do you have a list in mind of all the combinations mapped to what we have in the repo?

**Comment by anargyri** [comment]
> @anargyri do you have a list in mind of all the combinations mapped to what we have in the repo?

I was thinking of concatenating the labels we already had before the change with the ones you introduced. I.e. appending instead of overwriting.

File: README.md [hunk 1]
```diff
@@ -69,39 +69,39 @@ jupyter notebook
...
-| Extreme Deep Factorization Machine (xDeepFM)<sup>*</sup> | [Python CPU / Python GPU](examples/00_quick_start/xdeepfm_criteo.ipynb) | Hybrid | Deep learning based algorithm for implicit and explicit feedback with user/item features |
+| Convolutional Sequence Embedding Recommendation (Caser) | [Python CPU / Python GPU](examples/00_quick_start/sequential_recsys_amazondataset.ipynb) | Sequential | Algorithm based on convolutions that aim to capture both user’s general preferences and sequential patterns |
+| Deep Knowledge-Aware Network (DKN)<sup>*</sup> | [Python CPU / Python GPU](examples/00_quick_start/dkn_MIND.ipynb) | Knowledge based | Deep learning algorithm incorporating a knowledge graph and article embeddings to provide powerful news or article recommendations |
+| Extreme Deep Factorization Machine (xDeepFM)<sup>*</sup> | [Python CPU / Python GPU](examples/00_quick_start/xdeepfm_criteo.ipynb) | Content-Based Filtering | Deep learning based algorithm for implicit and explicit feedback with user/item features |
```
Comment (anargyri) [inline README.md:80]: xDeepFM is hybrid I think.

File: README.md [hunk 2]
```diff
@@ -69,39 +69,39 @@ jupyter notebook
...
 | FastAI Embedding Dot Bias (FAST) | [Python CPU / Python GPU](examples/00_quick_start/fastai_movielens.ipynb) | Collaborative Filtering | General purpose algorithm with embeddings and biases for users and items |
-| LightFM/Hybrid Matrix Factorization | [Python CPU](examples/02_model_hybrid/lightfm_deep_dive.ipynb) | Hybrid | Hybrid matrix factorization algorithm for both implicit and explicit feedbacks |
+| LightFM/Hybrid Matrix Factorization | [Python CPU](examples/02_model_hybrid/lightfm_deep_dive.ipynb) | Content-Based Filtering | Hybrid matrix factorization algorithm for both implicit and explicit feedbacks |
```
Comment (anargyri) [inline README.md:82]: This too is hybrid as the name suggests.

File: README.md [hunk 3]
```diff
@@ -69,39 +69,39 @@ jupyter notebook
...
 | LightGCN | [Python CPU / Python GPU](examples/02_model_collaborative_filtering/lightgcn_deep_dive.ipynb) | Collaborative Filtering | Deep learning algorithm which simplifies the design of GCN for predicting implicit feedback |
-| GeoIMC | [Python CPU](examples/00_quick_start/geoimc_movielens.ipynb) | Hybrid | Matrix completion algorithm that has into account user and item features using Riemannian conjugate gradients optimization and following a geometric approach. |
-| GRU4Rec | [Python CPU / Python GPU](examples/00_quick_start/sequential_recsys_amazondataset.ipynb) | Collaborative Filtering | Sequential-based algorithm that aims to capture both long and short-term user preferences using recurrent neural networks |
+| GeoIMC | [Python CPU](examples/00_quick_start/geoimc_movielens.ipynb) | Content-Based Filtering | Matrix completion algorithm that has into account user and item features using Riemannian conjugate gradients optimization and following a geometric approach. |
```
Comment (anargyri) [inline README.md:85]: This one is hybrid or CF.

File: README.md [hunk 4]
```diff
@@ -69,39 +69,39 @@ jupyter notebook
...
 | Vowpal Wabbit (VW)<sup>*</sup> | [Python CPU (online training)](examples/02_model_content_based_filtering/vowpal_wabbit_deep_dive.ipynb) | Content-Based Filtering | Fast online learning algorithms, great for scenarios where user features / context are constantly changing |
-| Wide and Deep | [Python CPU / Python GPU](examples/00_quick_start/wide_deep_movielens.ipynb) | Hybrid | Deep learning algorithm that can memorize feature interactions and generalize user features |
-| xLearn/Factorization Machine (FM) & Field-Aware FM (FFM) | [Python CPU](examples/02_model_hybrid/fm_deep_dive.ipynb) | Hybrid | Quick and memory efficient algorithm to predict labels with user/item features |
+| Wide and Deep | [Python CPU / Python GPU](examples/00_quick_start/wide_deep_movielens.ipynb) | Content-Based Filtering | Deep learning algorithm that can memorize feature interactions and generalize user features |
+| xLearn/Factorization Machine (FM) & Field-Aware FM (FFM) | [Python CPU](examples/02_model_hybrid/fm_deep_dive.ipynb) | Content-Based Filtering | Quick and memory efficient algorithm to predict labels with user/item features |
```
Comment (anargyri) [inline README.md:105]: FM and FFM are also hybrid (since they extend matrix factorization).

**Comment by miguelgfierro** [comment]
@anargyri about the hybrid mode, do you know some literature that we can use as a reference? In the book of Aggarwal there is a different definition of hybrid

**Comment by anargyri** [comment]
> @anargyri about the hybrid mode, do you know some literature that we can use as a reference? In the book of Aggarwal there is a different definition of hybrid

So, Aggarwal classifies methods as "collaborative filtering", "content-based filtering", "hybrid" etc. But "hybrid" is more general, it covers what we call "hybrid", but includes other approaches too.
The papers on which the notebooks are based reference these terms. For example, the Hybrid FM paper calls it a "hybrid method" explicitly.

Ideally we would like to follow the labels that the authors use for their own work. In fact, we could even contact the authors to make suggestions.

**Pull request closed without merging**